### PR TITLE
component-base: test and fix JSON output for KObj

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

KObj wasn't get logged as struct when using JSON output.

#### Does this PR introduce a user-facing change?
```release-note
Kubernetes object references (= name + namespace) were not logged as struct when using JSON as log output format.
```

/cc @serathius @thockin

### CODE REVIEW

File: staging/src/k8s.io/component-base/logs/json/klog_test.go [hunk 1]
```diff
@@ -116,6 +118,13 @@ func TestKlogIntegration(t *testing.T) {
 			},
 			format: `{"ts":%f,"caller":"json/klog_test.go:%d","msg":"test","v":0,"count":1}`,
 		},
+		{
+			name: "KObj",
```
Comment (serathius) [inline staging/src/k8s.io/component-base/logs/json/klog_test.go:122]: Can you also add a test for KObjs?

Comment (pohly) [reply staging/src/k8s.io/component-base/logs/json/klog_test.go:122]: Not yet - https://github.com/kubernetes/kubernetes/pull/104103 needs to be merged first.

But I've tested locally based on that PR and KObjs prints the slice as structs:
```
		{
			name: "KObjs",
			fun: func() {
				klog.InfoS("several", "pods",
					klog.KObjs([]interface{}{
						&kmeta{Name: "pod-1", Namespace: "kube-system"},
						&kmeta{Name: "pod-2", Namespace: "kube-system"},
					}))
			},
			format: `{"ts":%f,"caller":"json/klog_test.go:%d","msg":"several","v":0,"pods":[{"name":"pod-1","namespace":"kube-system"},{"name":"pod-2","namespace":"kube-system"}]}`,
		},
```

Comment (serathius) [reply staging/src/k8s.io/component-base/logs/json/klog_test.go:122]: heh, that makes sense as slice doesn't implement Stringer.

**Comment by serathius** [comment]
Sad that we missed this, I think I remember noticing this at some point, but when final implementation came we didn't have tests. We should definetly fix it for 1.23, @pohly can you file a issue to address this. We should track it in the Structured Logging KEP.

**Comment by pohly** [comment]
/hold

Let's use this PR to include a fix, if there is one.

**Pull request merged**